File: assignments/go/point-mutations/point-mutations_test.go
package pointmutations

import "testing"

var tests = []struct {
	expected int
	strand1  string
	strand2  string
}{
	{0, "", ""},               // empty
	{0, "GGACTGA", "GGACTGA"}, //identical
	{1, "GGACG", "GGTCG"},
	{2, "ACCAGGG", "ACTATGG"},
	{3, "AAACTAGGGG", "AGGCTAGCGGTAGGAC"},          // 2nd longer
	{5, "GACTACGGACAGGGTAGGGAAT", "GACATCGCACACC"}, // first longer
}

func TestHammingDistance(t *testing.T) {
	for _, test := range tests {
		actual := HammingDistance(test.strand1, test.strand2)
		if actual != test.expected {
			t.Errorf("HammingDistance(%s, %s): expected %d, actual %d", test.strand1, test.strand2, test.expected, actual)
		}
	}
}

File: assignments/elixir/point-mutations/point_mutations_test.exs
if System.get_env("EXERCISM_TEST_EXAMPLES") do
  Code.load_file("example.exs")
else
  Code.load_file("point_mutations.exs")
end

ExUnit.start

defmodule DNATest do
  use ExUnit.Case, async: true
  doctest DNA

  test "no difference between empty strands" do
    assert 0 == DNA.hamming_distance('', '')
  end

  test "no difference between identical strands" do
    # assert 0 == DNA.hamming_distance('GGACTGA', 'GGACTGA')
  end

  test "hamming distance in off by one strand" do
    # assert 19 == DNA.hamming_distance('GGACGGATTCTGACCTGGACTAATTTTGGGG', 'AGGACGGATTCTGACCTGGACTAATTTTGGGG')
  end

  test "small hamming distance in middle somewhere" do
    # assert 1 == DNA.hamming_distance('GGACG', 'GGTCG')
  end

  test "larger distance" do
    # assert 2 == DNA.hamming_distance('ACCAGGG', 'ACTATGG')
  end

  test "ignores extra length on other strand when longer" do
    # assert 3 == DNA.hamming_distance('AAACTAGGGG', 'AGGCTAGCGGTAGGAC')
  end

  test "ignores extra length on original strand when longer" do
    # assert 5 == DNA.hamming_distance('GACTACGGACAGGGTAGGGAAT', 'GACATCGCACACC')
  end
end
